File: go/src/leetcode/editor/cn/0163_missing_ranges.go
package main

import "strconv"

func main() {
	println("163、missing_ranges")
}

// leetcode submit region begin(Prohibit modification and deletion)
func findMissingRanges(nums []int, lower int, upper int) []string {
	r := make([]string, 0)
	nums = append([]int{lower - 1}, nums...)
	nums = append(nums, upper+1)
	for i := 1; i < len(nums); i++ {
		v := nums[i] - nums[i-1]
		if v == 2 {
			r = append(r, strconv.Itoa(nums[i]-1))
		}
		if v > 2 {
			r = append(r, strconv.Itoa(nums[i-1]+1)+"->"+strconv.Itoa(nums[i]-1))
		}
	}
	return r
}

//leetcode submit region end(Prohibit modification and deletion)

File: rust/src/leetcode/editor/cn/_0163_missing_ranges.rs
#[cfg(test)]
mod tests {
    #[allow(unused_imports)]
    use super::*;

    #[test]
    fn test() {
        println!("163、missing_ranges");
    }
}

#[allow(dead_code)]
struct Solution;

//leetcode submit region begin(Prohibit modification and deletion)
impl Solution {
    #[allow(dead_code)]
    pub fn find_missing_ranges(mut nums: Vec<i32>, lower: i32, upper: i32) -> Vec<String> {
        nums.insert(0, lower - 1);
        nums.push(upper + 1);
        let mut res = vec![];
        for i in 1..nums.len() {
            let value = nums[i] - nums[i - 1];
            if value == 2 {
                res.push((nums[i] - 1).to_string())
            } else if value > 2 {
                res.push(
                    (nums[i - 1] + 1).to_string() + &*"->".to_owned() + &*(nums[i] - 1).to_string(),
                )
            }
        }
        res
    }
}
//leetcode submit region end(Prohibit modification and deletion)
